### PR TITLE
Added unit-tests for UDP listener

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Added unit-tests in UTC listener.

### Motivation

Assert messages bigger than `4096` can be forwarded to the agent without being truncated.

### CODE REVIEW

File: pkg/logs/input/listener/udp_test.go [hunk 1]
```diff
@@ -38,28 +38,62 @@ func TestUDPShouldReceiveMessage(t *testing.T) {
...
+	fmt.Fprintf(conn, strings.Repeat("a", defaultFrameSize)+"\n")
+	msg = <-msgChan
+	assert.Equal(t, strings.Repeat("a", defaultFrameSize), string(msg.Content()))
+
+	fmt.Fprintf(conn, strings.Repeat("a", defaultFrameSize+100)+"\n")
 	msg = <-msgChan
-	assert.Equal(t, strings.Repeat("a", 80), string(msg.Content()))
+	assert.Equal(t, strings.Repeat("a", defaultFrameSize), string(msg.Content()))
+
+	fmt.Fprintf(conn, strings.Repeat("a", defaultFrameSize-200)+"\n")
```
Comment (tmichelet) [inline pkg/logs/input/listener/udp_test.go]: what's the difference between this test and the first one?

File: pkg/logs/input/listener/udp_test.go [hunk 2]
```diff
@@ -38,28 +38,62 @@ func TestUDPShouldReceiveMessage(t *testing.T) {
...
+
+	fmt.Fprintf(conn, strings.Repeat("a", 65535-100)+"\n")
 	msg = <-msgChan
-	assert.Equal(t, strings.Repeat("a", 100), string(msg.Content()))
+	assert.Equal(t, strings.Repeat("a", 65535-100), string(msg.Content()))
+
+	fmt.Fprintf(conn, strings.Repeat("a", 65535+100)+"\n")
+	select {
+	case <-msgChan:
+		assert.Fail(t, "expected message to be dropped")
```
Comment (tmichelet) [inline pkg/logs/input/listener/udp_test.go]: does this fail as expected when we update to a value lower than 65535?

File: pkg/logs/input/listener/udp_test.go (hunk 1 above)
Comment (ajacquemot) [reply pkg/logs/input/listener/udp_test.go]: it's just to ensure that the next message won't be mixed with the rest of the content of the previous message

Comment (tmichelet) [reply pkg/logs/input/listener/udp_test.go]: worth a comment then

**Review by tmichelet — DISMISSED** [review]

**Review by tmichelet — APPROVED** [review]

**Pull request merged**